Source/Core/Core/NetPlayClient.cpp

```diff
@@ -888,6 +1113,117 @@ bool NetPlayClient::StartGame(const std::string& path)
   return true;
 }
 
+void NetPlayClient::SyncSaveDataResponse(const bool success)
+{
+  m_dialog->AppendChat(success ? GetStringT("Data received!") :
+                                 GetStringT("Error processing data."));
+
+  if (success)
+  {
+    if (++m_sync_save_data_success_count >= m_sync_save_data_count)
+    {
+      sf::Packet response_packet;
+      response_packet << static_cast<MessageId>(NP_MSG_SYNC_SAVE_DATA);
+      response_packet << static_cast<MessageId>(SYNC_SAVE_DATA_SUCCESS);
+
+      Send(response_packet);
+    }
+  }
+  else
+  {
+    sf::Packet response_packet;
+    response_packet << static_cast<MessageId>(NP_MSG_SYNC_SAVE_DATA);
+    response_packet << static_cast<MessageId>(SYNC_SAVE_DATA_FAILURE);
+
+    Send(response_packet);
+  }
+}
+
+bool NetPlayClient::DecompressPacketIntoFile(sf::Packet& packet, const std::string& file_path)
+{
+  u32 size_low, size_high;
+  packet >> size_low >> size_high;
+  u64 file_size = size_low | (static_cast<u64>(size_high) << 32);
+
+  if (file_size == 0)
+    return true;
+
+  File::IOFile file(file_path, "wb");
```

Comment:
This should likely have a:

```cpp
if (!file)
  return false;
```

after it.

Reply:
Fixed.